addons/account/models/account_move_line.py

```diff
@@ -2333,32 +2333,22 @@ def remove_move_reconcile(self):
     # ANALYTIC
     # -------------------------------------------------------------------------
 
-    def _validate_distribution(self):
+    def _validate_analytic_distribution(self):
         for line in self.filtered(lambda line: line.display_type == 'product'):
-            mandatory_plans_ids = [plan['id'] for plan in self.env['account.analytic.plan'].sudo().get_relevant_plans(**{
+            line._validate_distribution(**{
```

Comment:
no need for the loop
`self.filtered(lambda line: line.display_type == 'product')._validate_distribution(**{...})`